libevmasm/PeepholeOptimiser.cpp

```diff
@@ -205,6 +205,31 @@ struct SwapComparison: SimplePeepholeOptimizerMethod<SwapComparison, 2>
 	}
 };
 
+/// Remove swapN after dupN
+struct DupSwap: SimplePeepholeOptimizerMethod<DupSwap, 2>
+{
+	static size_t applySimple(
+		AssemblyItem const& _dupN,
+		AssemblyItem const& _swapN,
+		std::back_insert_iterator<AssemblyItems> _out
+	)
+	{
+		if (
+			SemanticInformation::isDupInstruction(_dupN) &&
+			SemanticInformation::isSwapInstruction(_swapN) &&
+				uint8_t(_dupN.instruction()) - uint8_t(Instruction::DUP1) ==
+				uint8_t(_swapN.instruction()) - uint8_t(Instruction::SWAP1)
```

Comment:
Nitpick: this is a weird formatting with the extra indentation.

Reply:
It is correct according to our coding style due to the parenthesis around it... but yeah, it's a bit weird here :-).

This weird subtraction in general might be worth a helper in general, but maybe not, and I'd be fine with keeping it as is for this PR.

Reply:
There was no parenthesis the time I left this comment. Github still shows the original code:

![Screenshot 2020-07-13 at 13 44 12](https://user-images.githubusercontent.com/20340/87305775-f0e56180-c50e-11ea-9261-b52de9b944fe.png)